NitroxClient/Communication/Packets/Processors/PDAScanProgressProcessor.cs

```diff
@@ -0,0 +1,30 @@
+using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxClient.GameLogic;
+using NitroxModel.Packets;
+
+namespace NitroxClient.Communication.Packets.Processors;
+
+public class PDAScanProgressProcessor : ClientPacketProcessor<PDAScanProgress>
+{
+    public override void Process(PDAScanProgress packet)
+    {
+        // Fully-scanned entity
+        if (packet.Progress == 1f)
+        {
+            NitroxStoryManager.ScanCompleted(packet.Id, packet.Destroy);
+            return;
+        }
```

Comment:
Sounds good to me but the issue would only come from serialization/deserialization since we send a hard-coded 1f.

Reply:
Since it's a getter and has no state it will be ignored by serializers.

As for the 1f, UWE code does apply arithmetic to this field, which I assume is passed as is into the packet field:
```cs
PDAScanner.scanTarget.progress += Time.deltaTime / num1
```

Reply:
The game won't send any packet if the progress >= 1 (see the original method, the line right after the one you mentionned)
Here is the only place from which we send such a progress (PDAScanner_Scan_Patch.cs L79)
https://github.com/SubnauticaNitrox/Nitrox/pull/1967/files#diff-46f955dadb1f0044def5dfa7ea9fec0a5796f1964e01c1a147bd51276ac91af7R79
Considering this I shall revert the change because 0.99 is a valid value for this field and we don't want it to be confused with 1. If we expect this behaviour (sending 1f) as too "dangerous", we could still make it 10f and modify the processors accordingly.